irohad/ametsuchi/impl/postgres_query_executor.cpp

```diff
@@ -0,0 +1,800 @@
+      RIGHT OUTER JOIN has_perms ON TRUE
+      )")
+                  % hasQueryPermission(creator_id_,
+                                       q.accountId(),
+                                       Role::kGetMySignatories,
+                                       Role::kGetAllSignatories,
+                                       Role::kGetDomainSignatories))
+                     .str();
+      soci::statement st = (sql_.prepare << cmd);
+      st.exchange(soci::use(q.accountId()));
+      st.exchange(soci::into(row, ind));
+
+      st.define_and_bind();
+      st.execute();
+
+      int has_perm = -1;
+
+      processSoci(st,
+                  ind,
+                  row,
+                  [&pubkeys, &has_perm](
+                      boost::tuple<boost::optional<std::string>, int> &row) {
+                    has_perm = row.get<1>();
+                    if (row.get<0>()) {
+                      pubkeys.push_back(shared_model::crypto::PublicKey(
+                          shared_model::crypto::Blob::fromHexString(
+                              row.get<0>().get())));
+                    }
+                  });
+
+      if (not has_perm) {
+        return statefulFailed();
+      }
+
+      if (pubkeys.empty()) {
+        return buildError<
+            shared_model::interface::NoSignatoriesErrorResponse>();
+      }
+      return QueryResponseBuilder().signatoriesResponse(pubkeys);
+    }
+
+    QueryResponseBuilderDone PostgresQueryExecutorVisitor::operator()(
+        const shared_model::interface::GetAccountTransactions &q) {
+      soci::indicator ind;
+      using T = boost::
+          tuple<boost::optional<uint64_t>, boost::optional<uint64_t>, int>;
+      T row;
+      auto cmd = (boost::format(R"(WITH has_perms AS (%s),
+      t AS (
+          SELECT DISTINCT has.height, index
+          FROM height_by_account_set AS has
+          JOIN index_by_creator_height AS ich ON has.height = ich.height
+          AND has.account_id = ich.creator_id
+          WHERE account_id = :account_id
+          ORDER BY has.height, index ASC
+      )
+      SELECT height, index, perm FROM t
+      RIGHT OUTER JOIN has_perms ON TRUE
+      )")
+                  % hasQueryPermission(creator_id_,
+                                       q.accountId(),
+                                       Role::kGetMyAccTxs,
+                                       Role::kGetAllAccTxs,
+                                       Role::kGetDomainAccTxs))
+                     .str();
+      soci::statement st = (sql_.prepare << cmd);
+      st.exchange(soci::use(q.accountId()));
+      st.exchange(soci::into(row, ind));
+
+      st.define_and_bind();
+      st.execute();
+
+      int has_perm = -1;
+
+      std::map<uint64_t, std::vector<uint64_t>> index;
+
+      processSoci(st, ind, row, [&index, &has_perm](T &row) {
+        has_perm = row.get<2>();
+        if (row.get<0>()) {
+          if (index.find(row.get<0>().get()) == index.end()) {
```

Comment:
map creates an element if it not exists when it is accessed with `operator[]`, so this conditional is redundant.

Please check all the methods for the same issue.